git/cmd.py

```diff
@@ -603,6 +610,73 @@ def execute(self, command,
         else:
             return stdout_value
 
+    def set_environment(self, **kwargs):
```

Comment:
From looking at the implementation, I'd be tempted to call this method `update_environment`. `set_environment` seems to be more like this:
```python
res = self._environment
self._environment = kwargs
return res
```